src/Composer/DependencyResolver/PoolBuilder.php

```diff
@@ -43,7 +44,7 @@ class PoolBuilder
     private $io;
 
     private $aliasMap = array();
-    private $nameConstraints = array();
+    private $packagesToLoad = array();
     private $loadedNames = array();
```

Comment:
Just to make sure we dont mix up types

```suggestion
    /** @psalm-var array<string, Constraint>
    private $loadedNames = array();
```